opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -389,23 +390,19 @@ private String getFirstRevision(String fullpath) throws IOException {
                 ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK),
                 "rev-list",
                 "--reverse",
-                "--max-count=1",
                 "HEAD",
                 "--",
                 fullpath
         };
 
         Executor executor = new Executor(Arrays.asList(argv), new File(getDirectoryName()),
                 RuntimeEnvironment.getInstance().getInteractiveCommandTimeout());
-        int status = executor.exec();
-
-        try (BufferedReader in = new BufferedReader(
-                new InputStreamReader(executor.getOutputStream()))) {
-            String line;
+        HeadHandler headHandler = new HeadHandler(1);
+        int status = executor.exec(false, headHandler);
 
-            if ((line = in.readLine()) != null) {
-                return line.trim();
-            }
+        String line;
+        if ((line = headHandler.get(0)) != null) {
```

Comment:
I think the API requires `if (headHandler.count() > 0) {`

Reply:
I see, for some reason I thought it will return `null`.